docs/management-sdk-migration.md

```diff
@@ -0,0 +1,115 @@
+* The new APIs are more friendly for negotiation with multiple SignalR Service instances. `IServiceManager.GetClientEndpoint` and `IServiceManager.GenerateClientAccessToken` are combined into one method to make sure the client endpoint and the access token come from the same SignalR Service endpoint. An `HttpContext` instance is allowed to passed into the endpoint router to provide more information for the routing. <!--Todo Add link about sharding doc-->
+
+## Change Overview
+The following image shows the relation for the legacy APIs (green blocks) and the new APIs (blue blocks). Currently, to get a `ServiceHubContext` which implements `IServiceHubContext` interface, you just need to create it directly from `ServiceHubContextBuilder`. The `ServiceHubContext` abstract class combines the functionalities of `IServiceManager` and `IServiceHubContext`.
+
+![image](./images/migration-class-diagram.png)
+
+## API comparisons
+
+### Entry point
+
+**Legacy APIs**
+```cs
+var serviceManager = new ServiceManagerBuilder()
+    .WithOptions(o => o.ConnectionString = "<Your ConnectionString>")
+    .Build();
+```
+
+**New APIs**
+```cs
+var serviceHubContext = await new ServiceHubContextBuilder()
+    .WithOptions(o => o.ConnectionString = "<Your ConnectionString>")
+    .CreateAsync("<Your Hub Name>", cancellationToken);
+```
+<!--Add sharding link-->
+
+Note that you need to specify a hub name at the beginning in the new APIs.
+
+### Negotiate
+
+**Legacy APIs**
+```cs
+var clientEndpoint = serviceManager.GetClientEndpoint("<Your Hub Name>");
+var accessToken = serviceManager.GenerateClientAccessToken("<Your Hub Name>", "<Your User ID>");
+```
+
+**New APIs**
+```cs
+var negotiationResponse = await serviceHubContext.NegotiateAsync(new NegotiationOptions(){UserId = "<Your User Id>"});
+var clientEndpoint = negotiationResponse.Url;
+var accessToken = negotiationResponse.AccessToken;
+```
+The negotiation API is async now because there is an async operation behind if you use AAD connection string. In the old negotiation API, you just wait for the result synchronously.
+
+Except for the API change, there is also an implementation change. In [Persistent mode](management-sdk-guide.md#transport-type), you will call negotiation API after trying to establish WebSocket connections to Azure SignalR Service, so that you know whether the SignalR Service is healthy/online through the connection status. Therefore, the negotiation API only returns an healthy SignalR endpoint, if there is no healthy SignalR endpoint, an `AzureSignalRNotConnectedException` is thrown. In [Transient mode](management-sdk-guide.md#transport-type), we also have a plan to implement an endpoint health checker.
+
+### Send Messages and Manage Groups
+**Legacy APIs**
+```cs
+try
+{
+    var hubcontext = await serviceManager.CreateHubContextAsync("<Your Hub Name>");
+
+    // Broadcast
+    hubContext.Clients.All.SendAsync(callbackName, obj1, obj2, ...);
```

Comment:
await ...

Reply:
Haha I copied it from the previous guide and didn't notice this.